plugins/src/main/java/opengrok/auth/plugin/UserWhiteListPlugin.java

```diff
@@ -77,7 +90,13 @@ private boolean checkWhitelist(HttpServletRequest request) {
             return false;
         }
 
-        return whitelist.contains(user.getUsername());
+        if (fieldName.equals(USERNAME_FIELD)) {
+            return user.getUsername() != null && whitelist.contains(user.getUsername());
+        } else if (fieldName.equals(ID_FIELD)) {
+            return user.getId() != null && whitelist.contains(user.getId());
+        }
```

Comment:
I'd prefer enum, should be faster, could be arguably more readable

Reply:
I'd rather keep the configuration parameters of plugins restricted to strings. Or do you mean introduce the enum alongside the configuration only for internal handling in the plugin ?